xds/src/main/java/io/grpc/xds/XdsClusterSubscriptionRegistry.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2024 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.xds;
+
+public interface XdsClusterSubscriptionRegistry {
+  ClusterSubscription subscribeToCluster(String clusterName);
```

Comment:
This `ClusterSubscription` is pretty useless. If we didn't have it, the caller could just pass `clusterName` to `releaseSubscription()`. `ClusterSubscription` could be more useful if it avoided double-release, but it doesn't look like it's intended to.

I imagine we should either just pass `clusterName` to release, or change subscribe to return a Closable and have it detect double-close.

Reply:
This is not currently a subscription handle. I'm fine with multiple things subscribing to the same cluster.

Reply:
What is the GRPC definition of a "subscription handle"?  I assumed that it meant an object that had minimal functionality and acted as a representation of the particular subscription and whose only real use was passing to cancel.

Reply:
Changed to have `subscribeToCluster` return a Closeable which is much cleaner.